clowder-framework/clowder
# Extractor Catalog: Extractor Labels features

## Description
Labels to help organize and categorize the Extractor Catalog

Demonstration: http://recordit.co/zd0LTP6Fqu

### Problem
As the Extractor Catalog grows, it will become more and more difficult to search this list.

### Approach
Allow admin users to set arbitrary labels on Extractors, to help their users find them more easily.

## How to Test
Prerequisites: RabbitMQ plugin enabled, at least one extractor registered

1. Checkout and run this branch locally
2. Login to Clowder as an admin user
3. Navigate to Explore > Extractors
    * You should see a Labels section at the top of this view
4. Click the Manage button to the top-right of the Labels section
    * You should be brought to the Label Management view
    * NOTE: You can also reach this view by expanding the Admin dropdown and choosing "Extractor Labels"
5. Click on "Create New Label" fill out the following fields and click "Save":
    * Label Name (required)
    * Label Category Name (optional, defaults to "Other" if empty)
    * Assigned Extractors (optional)
    * You should see the modal is dismissed and the view is refreshed - your new label should now appear here
6. Click "Edit" beside your newly-created label to open the same modal again for editing
7. Change the assigned extractors, then click "Save" again
8. Navigate back to the Explore > Extractors view
    * You should see the Labels section contains a filter button for your newly-created / edited label
9. Click the filter button for your Label
    * You should see only the extractor you assigned previously appear in the table
10. Click the extractor's name in the table to see its ExtractorDetails view
    * You should see the Labels that have been added to the extractor are listed here
11. Add a `labels` section to your favorite `extractor_info.json` and register it via API or heartbeat.
    * For example: `  "labels":["Image","HelloWorld/DefaultLabel"]`
    * NOTE: These default labels in extractor_info.json can take the form "NAME" or "CATEGORY/NAME"
12. Navigate back to Admin > Extractor Labels and verify that our labels were imported properly
    * You should see that individual labels now exist for both Image and DefaultLabel
13. Click "Delete" beside an extractor label
    * You should see the view refresh and the label is now gone
    

## Review Time Estimate


## Types of changes


## Checklist:
